src/Agent.Sdk/Knob/ICompositeKnobSource.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Agent.Sdk.Knob
+{
+    public interface ICompositeKnobSource : IKnobSource
+    {
+        bool hasEnvironmentSourceWithName(string name);
```

Comment:
Name sounds little bit generic here (name of source - runtime/environment variable/build-in) or name of relevant variable (runtime/environment)?